src/py/_pyodide/_importhook.py

```diff
@@ -131,3 +131,41 @@ def register_js_finder() -> None:
     global JsProxy
     JsProxy = _pyodide_core.JsProxy
     sys.meta_path.append(jsfinder)
+
+
+class StdlibFinder(MetaPathFinder):
+    def __init__(self) -> None:
+        # `test`` is not a stdlib module, but we unvendors in anyway.
+        self.stdlibs = sys.stdlib_module_names | {"test"}
+
+        # TODO: put list of unvendored stdlibs to somewhere else?
+        self.unvendored_stdlibs = {"distutils", "test", "_ssl", "_lzma"} & self.stdlibs
+
+    def find_spec(
+        self,
+        fullname: str,
+        path: Sequence[bytes | str] | None,
+        target: ModuleType | None = None,
+    ) -> ModuleSpec | None:
+        [parent, _, _] = fullname.partition(".")
+
+        if not parent or parent in sys.modules or parent not in self.stdlibs:
+            return None
+
+        if parent in self.unvendored_stdlibs:
+            raise ModuleNotFoundError(
+                f"The module '{parent}' is unvendored from Pyodide stdlib, "
+                f'you can install it by calling: await pyodide.loadPackage("{parent}"). '
+                "See https://pyodide.org/en/stable/usage/wasm-constraints.html for more details."
+            )
+        else:
+            raise ModuleNotFoundError(
+                f"The module '{parent}' is removed from Pyodide stdlib "
+                "due to browser limitations. "
+                "See https://pyodide.org/en/stable/usage/wasm-constraints.html for more details."
+            )
+
+
+def register_stdlib_finder() -> None:
+    # Note: this finder must be placed in the end of meta paths.
+    sys.meta_path.append(StdlibFinder())
```

Comment:
Well we don't do that in `register_js_finder` plus it's private and should only be called once on initialization, so I am not sure that it is worth it. 

It is worth pointing out that the setup with where we call `register_js_finder` is a bit peculiar because of the specifics of the bootstrap process so we could call `register_stdlib_finder` in a more natural place. But keeping them together does make a lot of sense.

Reply:
Yeah, probably not necessary. It was just a thought.

Reply:
> Maybe check that's not already there before adding it, in case it's called twice for some reason?

Added a check.

> It is worth pointing out that the setup with where we call `register_js_finder` is a bit peculiar because of the specifics of the bootstrap process so we could call `register_stdlib_finder` in a more natural place. But keeping them together does make a lot of sense.

Yeah, it is not related to the bootstrap thing, but I just put it near the `regiest_js_finder`